closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeXDHKeyAgreement.java

```diff
@@ -0,0 +1,320 @@
+        secret = null;
+
+        if (!NativeCrypto.isAllowedAndLoaded()) {
+            if (nativeCryptTrace) {
+                System.err.println("OpenSSL library not loaded." +
+                    " Using Java crypto implementation to generate secret.");
+            }
+            initializeJavaImplementation();
+        } else {
+            /* Assign ID used by OpenSSL for different curves. */
+            int curveType;
+            if ("X25519".equalsIgnoreCase(ops.getParameters().getName())) {
+                curveType = NativeCrypto.X25519;
+            } else {
+                curveType = NativeCrypto.X448;
+            }
+
+            if ((curveType == NativeCrypto.X448)
+            && (NativeCrypto.getVersionIfAvailable() < NativeCrypto.OPENSSL_VERSION_3_0_0)
+            ) {
+                if (nativeCryptTrace) {
+                    System.err.println("OpenSSL version too old for X448 key agreement (<3.x)," +
+                        " using Java crypto implementation.");
+                }
+                initializeJavaImplementation();
+            }
+        }
+    }
+
+    @Override
+    protected Key engineDoPhase(Key key, boolean lastPhase)
+            throws InvalidKeyException, IllegalStateException {
+        byte[] computedSecret;
+        if (this.javaImplementation != null) {
+            computedSecret = javaImplComputeSecret(key, lastPhase);
+        } else {
+            if (this.privateKey == null) {
+                throw new IllegalStateException("Not initialized");
+            }
+            if (this.secret != null) {
+                throw new IllegalStateException("Phase already executed");
+            }
+            if (!lastPhase) {
+                throw new IllegalStateException(
+                    "Only two party agreement supported, lastPhase must be true");
+            }
+
+            if (!(key instanceof XDHPublicKeyImpl)) {
+                throw new InvalidKeyException("Unsupported key type");
+            }
+
+            XDHPublicKeyImpl publicKey = (XDHPublicKeyImpl) key;
+
+            // Ensure public key parameters are compatible with private key
+            XECParameters xecParams = XECParameters.get(InvalidKeyException::new,
+                publicKey.getParams());
+            if (!ops.getParameters().oidEquals(xecParams)) {
+                throw new InvalidKeyException(
+                    "Public key parameters are not compatible with private key.");
+            }
+
+            /* Assign ID used by OpenSSL for different curves */
+            int curveType;
+            if ("X25519".equalsIgnoreCase(ops.getParameters().getName())) {
+                curveType = NativeCrypto.X25519;
+            } else {
+                curveType = NativeCrypto.X448;
+            }
+
+            byte[] publicKeyArray = publicKey.getKeyAsByteArray();
+            if (publicKeyArray == null) {
+                throw new InvalidKeyException("No public key value");
+            }
```

Comment:
This can never happen; please just delete this `if` statement.

Reply:
Deleted.